elastic/beats
# Add ECS logging MVP to libbeat

## What does this PR do?
Enable ECS conformant logging by setting `logging.ecs:true`
Adds @timestamp, log.level, message and ecs.version to every log line.
If EnableCaller is set add log.origin.file and log.origin.line.
Logs error.message when error is given




## Why is it important?
Get started with ECS conformant logging



## Checklist



~- [ ] I have commented my code, particularly in hard-to-understand areas~

## How to test this PR locally
Set different logging options (syslog, to_files, console) and enable `logging.json` and `logging.ecs` and check log lines contain `@timestamp, message, log.level, ecs.version`. Logged errors are logged in the format `{ "error": { "message": .. } }`. If caller is enabled it is logged as `log.origin.*`. 

manually tested for 
* `logging.to_files: true`
* `logging.to_syslog: true`
* `logging.to_stderr: true`
* log output to console `./beat -e`




## Related issues

implements elastic/beats#17942



## Logs
Example:
```
{"log.level":"error","@timestamp":"2020-04-24T16:03:13.921+0200","log.origin":{"file.name":"instance/metrics.go","file.line":92},"message":"Error while getting memory usage: error retrieving process stats: cannot find matching process for pid=13762","ecs.version":"1.5.0"}
```
